lib/uri/common.rb

```diff
@@ -342,17 +342,17 @@ def self.regexp(schemes = nil)
 
   TBLENCWWWCOMP_ = {} # :nodoc:
   256.times do |i|
-    TBLENCWWWCOMP_[i.chr] = '%%%02X' % i
+    TBLENCWWWCOMP_[-i.chr] = -('%%%02X' % i)
   end
-  TBLENCWWWCOMP_[' '] = '+'
+  TBLENCWWWCOMP_[' '] = -'+'
```

Comment:
Why do you call `String#-@` in addition to specifying `# frozen_string_literal: true`? The `'+'` literal should be frozen and does no allocation without calling it. All other `String#-@` calls seem lengthy too.